### DISCUSSION TITLE
Define minimum node version for a package

### DISCUSSION THREAD

**Discussion by Torniojaws** [npm/feedback#421] [General]
I often come across severe version incompatibilities when updating old projects to a newer node version, and have to spend a humongous amount of time checking whether a package is compatible or not. Say, going from a project done with node 8.x being updated to node 14.x.

Could `package.json` have some system for this? For example

```
{
  "dependencies": {
    "some-package": {
       "version": "^1.2.0",
       "node": "<=10.0.0"
    },
    "another-package": {
       "version": "^2.0.0",
       "node": ">=12.0.0"
    },
    "different-package": {
       "version": "^4.5.0",
       "node": ">=12.0.0 <14.0.0"
    }
  }
}
```

Then it would become obvious that I would have to upgrade `some-package` and `different-package`, but `another-package` theoretically *should* still work.

**Comment by ljharb** [comment]
Each package, including your app, should be defining engines.node. Being able to set a deoendency’s engines declaration in your own package.json seems like it would create multiple sources of truth.
